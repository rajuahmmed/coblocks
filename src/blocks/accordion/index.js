/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	attributes,
	transforms,
	edit,
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
};

export { name, category, metadata, settings };
