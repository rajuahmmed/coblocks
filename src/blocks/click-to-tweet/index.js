/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import deprecated from './deprecated';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Click to Tweet', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a quote for readers to tweet via Twitter.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'share', 'coblocks' ),
		/* translators: block keyword */
		__( 'twitter', 'coblocks' ),
	],
	example: {
		attributes: {
			content: __( 'The easiest way to promote and advertise your blog, website, and business on Twitter.', 'coblocks' ),
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
