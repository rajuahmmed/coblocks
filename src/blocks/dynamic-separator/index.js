/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Dynamic HR', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a resizable spacer between other blocks.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		'hr',
		/* translators: block keyword */
		__( 'spacer', 'coblocks' ),
	],
	styles: [
		{
			name: 'dots',
			/* translators: block style */
			label: __( 'Dot', 'coblocks' ),
			isDefault: true,
		},
		{
			name: 'line',
			/* translators: block style */
			label: __( 'Line', 'coblocks' ),
		},
		{
			name: 'fullwidth',
			/* translators: block style */
			label: __( 'Fullwidth', 'coblocks' ),
		},
	],
	example: {
		attributes: {
			height: 100,
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
