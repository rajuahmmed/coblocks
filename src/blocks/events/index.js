/**
 * Internal dependencies.
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies.
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	title: _x( 'Events', 'block name', 'coblocks' ),
	description: __( 'Add a list of events or display events from a public calendar.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'calendar', 'coblocks' ),
		/* translators: block keyword */
		__( 'date', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
	},
	transforms,
	edit,
	save,
};

export { metadata, name, category, settings };
