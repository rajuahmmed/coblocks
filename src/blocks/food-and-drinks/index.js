/**
 * Internal dependencies.
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import deprecated from './deprecated';

/**
 * WordPress dependencies.
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Food & Drinks', 'coblocks' ),
	/* translators: block description */
	description: __( 'Display a menu or price list.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'restaurant', 'coblocks' ),
		/* translators: block keyword */
		__( 'menu', 'coblocks' ),
	],
	supports: {
		align: [ 'wide' ],
	},
	getEditWrapperProps( attributes ) {
		return { 'data-columns': attributes.columns };
	},
	deprecated,
	attributes,
	edit,
	save,
};

export { name, category, metadata, settings };
