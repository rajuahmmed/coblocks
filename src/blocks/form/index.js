/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import variations from './variations';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const metadata = {
	name: 'coblocks/form',
	category: 'layout',
	attributes: {
		subject: {
			type: 'string',
			default: null,
		},
		to: {
			type: 'string',
			default: null,
		},
	},
};

const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Form', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a contact form to your page.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'email', 'coblocks' ),
		/* translators: block keyword */
		__( 'about', 'coblocks' ),
		/* translators: block keyword */
		__( 'contact', 'coblocks' ),
	],
	supports: {
		reusable: false,
		html: false,
		customClassName: false,
	},
	example: {
		attributes: {
			subject: __( 'Subject example', 'coblocks' ),
		},
	},
	attributes,
	variations,
	edit,
	save: InnerBlocks.Content,
};

export { name, category, metadata, settings };
