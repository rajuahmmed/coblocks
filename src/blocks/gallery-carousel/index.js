/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import { GalleryAttributes } from '../../components/block-gallery/shared';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...GalleryAttributes,
	...metadata.attributes,
};

const settings = {
	/* translators: block name */
	title: __( 'Carousel', 'coblocks' ),
	/* translators: block description */
	description: __( 'Display multiple images in a beautiful carousel gallery.', 'coblocks' ),
	attributes,
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'gallery', 'coblocks' ),
		/* translators: block keyword */
		__( 'photos', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
		coBlocksSpacing: true,
	},
	example: {
		attributes: {
			gridSize: 'lrg',
			gutter: 5,
			images: [
				{ url: 'https://s.w.org/images/core/5.3/Biologia_Centrali-Americana_-_Cantorchilus_semibadius_1902.jpg' },
				{ url: 'https://s.w.org/images/core/5.3/Glacial_lakes,_Bhutan.jpg' },
				{ url: 'https://s.w.org/images/core/5.3/MtBlanc1.jpg' },
			],
		},
	},
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, icon, metadata, settings };
