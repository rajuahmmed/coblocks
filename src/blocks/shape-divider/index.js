/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Shape Divider', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a shape divider to visually distinquish page sections.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		'hr',
		'svg',
		/* translators: block keyword */
		__( 'separator', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		coBlocksSpacing: true,
	},
	styles: [
		{
			name: 'wavy',
			/* translators: block style */
			label: __( 'Wavy', 'coblocks' ),
			isDefault: true,
		},
		{
			name: 'hills',
			/* translators: block style */
			label: __( 'Hills', 'coblocks' ),
		},
		{
			name: 'waves',
			/* translators: block style */
			label: __( 'Waves', 'coblocks' ),
		},
		{
			name: 'angled',
			/* translators: block style */
			label: __( 'Angled', 'coblocks' ),
		},
		{
			name: 'sloped',
			/* translators: block style */
			label: __( 'Sloped', 'coblocks' ),
		},
		{
			name: 'rounded',
			/* translators: block style */
			label: __( 'Rounded', 'coblocks' ),
		},
		{
			name: 'triangle',
			/* translators: block style */
			label: __( 'Triangle', 'coblocks' ),
		},
		{
			name: 'pointed',
			/* translators: block style */
			label: __( 'Pointed', 'coblocks' ),
		},
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
