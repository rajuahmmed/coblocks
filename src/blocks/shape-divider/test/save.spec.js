/**
 * External dependencies
 */
import { JSDOM } from 'jsdom';
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let blockDOM;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	afterEach( () => {
		// Make a snapshot for each save function test to better detect deprecation needs.
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
	} );

	it( 'should have \'min-height\' property within the \'style\' attribute with the shapeHeight attribute set.', () => {
		block.attributes = { ...block.attributes, shapeHeight: 200 };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider__svg-wrapper' )[ 0 ]
		).toHaveStyle( 'min-height: 200px' );
	} );

	it( 'should have \'min-height\' property within the \'style\' attribute with the backgroundHeight attribute set.', () => {
		block.attributes = { ...block.attributes, backgroundHeight: 200 };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider__alt-wrapper' )[ 0 ]
		).toHaveStyle( 'min-height: 200px' );
	} );

	it( 'should have className \'is-vertically-flipped\' with verticalFlip enabled.', () => {
		block.attributes = { ...block.attributes, verticalFlip: false };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider' )[ 0 ]
		).not.toHaveClass( 'is-vertically-flipped' );

		block.attributes = { ...block.attributes, verticalFlip: true };
		serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider' )[ 0 ]
		).toHaveClass( 'is-vertically-flipped' );
	} );

	it( 'should have className \'is-horizontally-flipped\' with horizontalFlip enabled.', () => {
		block.attributes = { ...block.attributes, horizontalFlip: false };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider' )[ 0 ]
		).not.toHaveClass( 'is-horizontally-flipped' );

		block.attributes = { ...block.attributes, horizontalFlip: true };
		serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.getElementsByClassName( 'wp-block-coblocks-shape-divider' )[ 0 ]
		).toHaveClass( 'is-horizontally-flipped' );
	} );
} );
