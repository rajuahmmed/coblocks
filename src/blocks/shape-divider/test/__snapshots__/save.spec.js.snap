// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/shape-divider should have 'min-height' property within the 'style' attribute with the backgroundHeight attribute set. 1`] = `
"<!-- wp:coblocks/shape-divider {\\"backgroundHeight\\":200} -->
<div class=\\"wp-block-coblocks-shape-divider alignfull\\" style=\\"color:#111\\" aria-hidden=\\"true\\"><div class=\\"wp-block-coblocks-shape-divider__svg-wrapper\\" style=\\"min-height:100px\\"><svg class=\\"divider--wavy\\" viewbox=\\"0 0 100 10\\" xmlns=\\"http://www.w3.org/2000/svg\\" preserveaspectratio=\\"none\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m42.19.65c2.26-.25 5.15.04 7.55.53 2.36.49 7.09 2.35 10.05 3.57 7.58 3.22 13.37 4.45 19.26 4.97 2.36.21 4.87.35 10.34-.25s10.62-2.56 10.62-2.56v-6.91h-100.01v3.03s7.2 3.26 15.84 3.05c3.92-.07 9.28-.67 13.4-2.24 2.12-.81 5.22-1.82 7.97-2.42 2.72-.63 3.95-.67 4.98-.77z\\" fillrule=\\"evenodd\\" transform=\\"matrix(1 0 0 -1 0 10)\\"></path></svg></div><div class=\\"wp-block-coblocks-shape-divider__alt-wrapper\\" style=\\"min-height:200px\\"></div></div>
<!-- /wp:coblocks/shape-divider -->"
`;

exports[`coblocks/shape-divider should have 'min-height' property within the 'style' attribute with the shapeHeight attribute set. 1`] = `
"<!-- wp:coblocks/shape-divider {\\"shapeHeight\\":200} -->
<div class=\\"wp-block-coblocks-shape-divider alignfull\\" style=\\"color:#111\\" aria-hidden=\\"true\\"><div class=\\"wp-block-coblocks-shape-divider__svg-wrapper\\" style=\\"min-height:200px\\"><svg class=\\"divider--wavy\\" viewbox=\\"0 0 100 10\\" xmlns=\\"http://www.w3.org/2000/svg\\" preserveaspectratio=\\"none\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m42.19.65c2.26-.25 5.15.04 7.55.53 2.36.49 7.09 2.35 10.05 3.57 7.58 3.22 13.37 4.45 19.26 4.97 2.36.21 4.87.35 10.34-.25s10.62-2.56 10.62-2.56v-6.91h-100.01v3.03s7.2 3.26 15.84 3.05c3.92-.07 9.28-.67 13.4-2.24 2.12-.81 5.22-1.82 7.97-2.42 2.72-.63 3.95-.67 4.98-.77z\\" fillrule=\\"evenodd\\" transform=\\"matrix(1 0 0 -1 0 10)\\"></path></svg></div><div class=\\"wp-block-coblocks-shape-divider__alt-wrapper\\" style=\\"min-height:50px\\"></div></div>
<!-- /wp:coblocks/shape-divider -->"
`;

exports[`coblocks/shape-divider should have className 'is-horizontally-flipped' with horizontalFlip enabled. 1`] = `
"<!-- wp:coblocks/shape-divider {\\"horizontalFlip\\":true} -->
<div class=\\"wp-block-coblocks-shape-divider alignfull is-horizontally-flipped\\" style=\\"color:#111\\" aria-hidden=\\"true\\"><div class=\\"wp-block-coblocks-shape-divider__svg-wrapper\\" style=\\"min-height:100px\\"><svg class=\\"divider--wavy\\" viewbox=\\"0 0 100 10\\" xmlns=\\"http://www.w3.org/2000/svg\\" preserveaspectratio=\\"none\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m42.19.65c2.26-.25 5.15.04 7.55.53 2.36.49 7.09 2.35 10.05 3.57 7.58 3.22 13.37 4.45 19.26 4.97 2.36.21 4.87.35 10.34-.25s10.62-2.56 10.62-2.56v-6.91h-100.01v3.03s7.2 3.26 15.84 3.05c3.92-.07 9.28-.67 13.4-2.24 2.12-.81 5.22-1.82 7.97-2.42 2.72-.63 3.95-.67 4.98-.77z\\" fillrule=\\"evenodd\\" transform=\\"matrix(1 0 0 -1 0 10)\\"></path></svg></div><div class=\\"wp-block-coblocks-shape-divider__alt-wrapper\\" style=\\"min-height:50px\\"></div></div>
<!-- /wp:coblocks/shape-divider -->"
`;

exports[`coblocks/shape-divider should have className 'is-vertically-flipped' with verticalFlip enabled. 1`] = `
"<!-- wp:coblocks/shape-divider {\\"verticalFlip\\":true} -->
<div class=\\"wp-block-coblocks-shape-divider alignfull is-vertically-flipped\\" style=\\"color:#111\\" aria-hidden=\\"true\\"><div class=\\"wp-block-coblocks-shape-divider__svg-wrapper\\" style=\\"min-height:100px\\"><svg class=\\"divider--wavy\\" viewbox=\\"0 0 100 10\\" xmlns=\\"http://www.w3.org/2000/svg\\" preserveaspectratio=\\"none\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m42.19.65c2.26-.25 5.15.04 7.55.53 2.36.49 7.09 2.35 10.05 3.57 7.58 3.22 13.37 4.45 19.26 4.97 2.36.21 4.87.35 10.34-.25s10.62-2.56 10.62-2.56v-6.91h-100.01v3.03s7.2 3.26 15.84 3.05c3.92-.07 9.28-.67 13.4-2.24 2.12-.81 5.22-1.82 7.97-2.42 2.72-.63 3.95-.67 4.98-.77z\\" fillrule=\\"evenodd\\" transform=\\"matrix(1 0 0 -1 0 10)\\"></path></svg></div><div class=\\"wp-block-coblocks-shape-divider__alt-wrapper\\" style=\\"min-height:50px\\"></div></div>
<!-- /wp:coblocks/shape-divider -->"
`;

exports[`coblocks/shape-divider should render 1`] = `
"<!-- wp:coblocks/shape-divider -->
<div class=\\"wp-block-coblocks-shape-divider alignfull\\" style=\\"color:#111\\" aria-hidden=\\"true\\"><div class=\\"wp-block-coblocks-shape-divider__svg-wrapper\\" style=\\"min-height:100px\\"><svg class=\\"divider--wavy\\" viewbox=\\"0 0 100 10\\" xmlns=\\"http://www.w3.org/2000/svg\\" preserveaspectratio=\\"none\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m42.19.65c2.26-.25 5.15.04 7.55.53 2.36.49 7.09 2.35 10.05 3.57 7.58 3.22 13.37 4.45 19.26 4.97 2.36.21 4.87.35 10.34-.25s10.62-2.56 10.62-2.56v-6.91h-100.01v3.03s7.2 3.26 15.84 3.05c3.92-.07 9.28-.67 13.4-2.24 2.12-.81 5.22-1.82 7.97-2.42 2.72-.63 3.95-.67 4.98-.77z\\" fillrule=\\"evenodd\\" transform=\\"matrix(1 0 0 -1 0 10)\\"></path></svg></div><div class=\\"wp-block-coblocks-shape-divider__alt-wrapper\\" style=\\"min-height:50px\\"></div></div>
<!-- /wp:coblocks/shape-divider -->"
`;
